join.h

```diff
@@ -33,83 +38,105 @@ class Join {
      * eg: Consider the tables "Person" and "Worked" as below:
      * Person id | name                     Worked  id_company | id_person
      *         9 | Jhoe  (position 111)                  77    |     9        (position 555)
-     *        10 | Marta (position 222)                  35    |    10        (position 666)       
+     *        10 | Marta (position 222)                  35    |    10        (position 666)
      *                                                   44    |    10        (position 777)
      *
      * In this case, the result would be a vector below:
-     * [ [111,555], [222,666],[222,777]] 
+     * [ [111,555], [222,666],[222,777]]
      *
      *
      * @constructor
      */
-    Join(Queryable *this_table, string this_column_name, Queryable* other_table, string other_column_name,  JoinType join_type);
-    
+    Join(Queryable *this_table, string this_column_name, Queryable* other_table, string other_column_name, JoinType join_type, vector<long> *this_table_ids);
+
     /**
      * @destructor
      */
     ~Join();
-    
+
 
      /**
-     * Print the result of the join, converting the row of registries position into 
+     * Print the result of the join, converting the row of registries position into
      * the respective registries (for debugging only)
      * @param number_of_values the number of values to print. If set to -1
      *        all the file wil be printed
      */
     void print(int number_of_values);
 };
-vector<vector<long long>>* Join::nestedIndexJoin(Queryable *this_table, int this_column_position, Queryable* other_table, int other_column_position){
+vector<vector<long long>>* Join::nestedIndexLoopJoin(Queryable *this_table, int this_column_position, vector<long> *this_table_ids, Queryable *other_table, int other_column_position){
+    this->join_result = new vector<vector<long long>>;
+
+    for(int i=0; i<this_table_ids->size(); i++){
```

Comment:
Remove o argumento this_table_ids e itera sobre o this_table->getHeader()

for(header_t::iterator it = header->begin(); it != header->end(); it++) {
   long long id = it->first;
   long long offset = it->second;